src/Umbraco.Web.UI.Client/src/views/common/overlays/linkpicker/linkpicker.controller.js

```diff
@@ -1,192 +1,185 @@
-
-                // is it a content link?
-                if (!$scope.model.target.isMedia) {
-                    // get the content path
-                    entityResource.getPath(id, "Document").then(function (path) {
-                        //now sync the tree to this path
-                        $scope.dialogTreeEventHandler.syncTree({
-                            path: path,
-                            tree: "content"
-                        });
+angular.module("umbraco").controller("Umbraco.Overlays.LinkPickerController", function ($scope, eventsService, entityResource, contentResource, mediaHelper, userService, localizationService, tinyMceService) {
+    if (!$scope.model.title) {
+        $scope.model.title = localizationService.localize("defaultdialogs_selectLink");
+    }
+
+    $scope.placeholders_entername = localizationService.localize("placeholders_entername");
+
+    $scope.dialogTreeEventHandler = $({});
+    $scope.model.target = {};
+    $scope.searchInfo = {
+        searchFromId: null,
+        searchFromName: null,
+        showSearch: false,
+        ignoreUserStartNodes: $scope.model.ignoreUserStartNodes,
+        results: [],
+        selectedSearchResults: []
+    };
+    $scope.customTreeParams = $scope.model.ignoreUserStartNodes ? "ignoreUserStartNodes=" + $scope.model.ignoreUserStartNodes : "";
+    $scope.showTarget = $scope.model.hideTarget !== true;
+
+    if ($scope.model.currentTarget) {
+        // clone the current target so we don't accidentally update the caller's model while manipulating $scope.model.target
+        $scope.model.target = angular.copy($scope.model.currentTarget);
+
+        // if we have a node ID, we fetch the current node to build the form data
+        if ($scope.model.target.id || $scope.model.target.udi) {
+            // will be either a udi or an int
+            var id = $scope.model.target.udi ? $scope.model.target.udi : $scope.model.target.id;
+
+            // is it a content link?
+            if (!$scope.model.target.isMedia) {
+                // get the content path
+                entityResource.getPath(id, "Document").then(function (path) {
+                    // now sync the tree to this path
+                    $scope.dialogTreeEventHandler.syncTree({
+                        path: path,
+                        tree: "content"
                     });
+                });
 
-                    // if a link exists, get the properties to build the anchor name list
-                    contentResource.getById(id, { ignoreUserStartNodes: dialogOptions.ignoreUserStartNodes }).then(function (resp) {
-                        $scope.model.target.url = resp.urls[0];
-                        $scope.anchorValues = tinyMceService.getAnchorNames(JSON.stringify(resp.properties));
-                    });
-                }
-            } else if ($scope.model.target.url.length) {
```

Comment:
As `$scope.model.target.url` could be `null` or `undefined`, this could throw errors...